src/Kentico.Xperience.Algolia/Indexing/DefaultAlgoliaClient.cs

```diff
@@ -0,0 +1,231 @@
+using CMS.Websites;
+using Kentico.Xperience.Algolia.Admin;
+using Newtonsoft.Json.Linq;
+
+namespace Kentico.Xperience.Algolia.Indexing;
+
+/// <summary>
+/// Default implementation of <see cref="IAlgoliaClient"/>.
+/// </summary>
+internal class DefaultAlgoliaClient : IAlgoliaClient
+{
+    private readonly IAlgoliaIndexService algoliaIndexService;
+    private readonly IInfoProvider<ContentLanguageInfo> languageProvider;
+    private readonly IInfoProvider<ChannelInfo> channelProvider;
+    private readonly IConversionService conversionService;
+    private readonly IContentQueryExecutor executor;
+    private readonly IProgressiveCache cache;
+    private readonly ISearchClient searchClient;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DefaultAlgoliaClient"/> class.
+    /// </summary>
+    public DefaultAlgoliaClient(
+        IAlgoliaIndexService algoliaIndexService,
+        IInfoProvider<ContentLanguageInfo> languageProvider,
+        IInfoProvider<ChannelInfo> channelProvider,
+        IConversionService conversionService,
+        IProgressiveCache cache,
+        ISearchClient searchClient,
+        IContentQueryExecutor executor)
+    {
+        this.algoliaIndexService = algoliaIndexService;
+        this.cache = cache;
+        this.searchClient = searchClient;
+        this.executor = executor;
+        this.languageProvider = languageProvider;
+        this.channelProvider = channelProvider;
+        this.conversionService = conversionService;
+    }
+
+    /// <inheritdoc />
+    public Task<int> DeleteRecords(IEnumerable<string> itemGuids, string indexName, CancellationToken cancellationToken)
+    {
+        if (string.IsNullOrEmpty(indexName))
+        {
+            throw new ArgumentNullException(nameof(indexName));
+        }
+
+        if (itemGuids == null || !itemGuids.Any())
+        {
+            return Task.FromResult(0);
+        }
+
+        return DeleteRecordsInternal(itemGuids, indexName, cancellationToken);
+    }
+
+    /// <inheritdoc/>
+    private async Task<ICollection<IndicesResponse>> GetStatisticsInternal(CancellationToken cancellationToken) =>
+        (await searchClient.ListIndicesAsync(ct: cancellationToken).ConfigureAwait(false)).Items;
+
+    /// <inheritdoc/>
+    public async Task<ICollection<AlgoliaIndexStatisticsViewModel>> GetStatistics(CancellationToken cancellationToken) =>
+        (await GetStatisticsInternal(cancellationToken))
+        .Select(i => new AlgoliaIndexStatisticsViewModel
+        {
+            Name = i.Name,
+            Entries = i.Entries,
+            UpdatedAt = i.UpdatedAt
+        })
+        .ToList();
+
+    /// <inheritdoc />
+    public Task Rebuild(string indexName, CancellationToken cancellationToken)
+    {
+        if (string.IsNullOrEmpty(indexName))
+        {
+            throw new ArgumentNullException(nameof(indexName));
+        }
+
+        var algoliaIndex = AlgoliaIndexStore.Instance.GetIndex(indexName);
```

Comment:
```suggestion
        var algoliaIndex = AlgoliaIndexStore.Instance.GetRequiredIndex(indexName);
```

`GetRequiredIndex` will throw the exception.

Reply:
changed